src/unistd.rs

```diff
@@ -869,9 +881,15 @@ pub fn pipe2(flags: OFlag) -> Result<(RawFd, RawFd)> {
     Ok((fds[0], fds[1]))
 }
 
-#[cfg(not(any(target_os = "linux",
-              target_os = "android",
-              target_os = "emscripten")))]
+/// Like `pipe`, but allows setting certain file descriptor flags.
+///
+/// The following flags are supported, and will be set after the pipe is
+/// created:
+///
+/// `O_CLOEXEC`:    Set the close-on-exec flag for the new file descriptors.
+/// `O_NONBLOCK`:   Set the non-blocking flag for the ends of the pipe.
+#[cfg(any(target_os = "ios",
+          target_os = "macos"))]
 pub fn pipe2(flags: OFlag) -> Result<(RawFd, RawFd)> {
```

Comment:
Does it make sense to keep this emulation for just ios/macos? We've removed emulated functionality before, and I'd like to suggest we remove it here as well.

Reply:
Removing it would make me happy.  But perhaps we should mark it as deprecated for a release cycle first?

Reply:
Sure, let's mark it deprecated for 0.10 and then remove for the next release.